cpp/src/Group.cpp

```diff
@@ -200,18 +223,27 @@ bool Group::Contains
 //-----------------------------------------------------------------------------
 void Group::AddAssociation
 (
-	uint8 const _nodeId
+	uint8 const _nodeId,
+	uint8 const _instance
 )
 {
 	if( Driver* driver = Manager::Get()->GetDriver( m_homeId ) )
 	{
 		if( Node* node = driver->GetNodeUnsafe( m_nodeId ) )
 		{
-			if( Association* cc = static_cast<Association*>( node->GetCommandClass( Association::StaticGetCommandClassId() ) ) )
+			if( MultiInstanceAssociation* cc = static_cast<MultiInstanceAssociation*>( node->GetCommandClass( MultiInstanceAssociation::StaticGetCommandClassId() )))
 			{
-				cc->Set( m_groupIdx, _nodeId );
+				cc->Set( m_groupIdx, _nodeId, _instance );
 				cc->QueryGroup( m_groupIdx, 0 );
 			}
+			else
```

Comment:
this could be a else if and the final else, I would log a message about no supported Association CC's found.